test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -420,6 +421,50 @@ public void CanUpgradeRequestWithConnectionKeepAliveUpgradeHeader()
             }
         }
 
+        [Theory]
+        [InlineData("http://localhost/abs/path", "/abs/path")]
+        [InlineData("https://localhost/abs/path", "/abs/path")] // handles mismatch scheme
+        [InlineData("https://localhost:22/abs/path", "/abs/path")] // handles mismatched ports
+        [InlineData("https://differenthost/abs/path", "/abs/path")] // handles mismatched hostname
+        [InlineData("http://localhost/", "/")]
+        [InlineData("http://root@contoso.com/path", "/path")]
+        [InlineData("http://root:password@contoso.com/path", "/path")]
+        [InlineData("https://localhost/", "/")]
+        [InlineData("http://localhost", "")]
```

Comment:
I'm favor of keeping it and setting to "/". The HTTP spec says is `absolute-form = absolute-URI`, and absolute URI's don't require the trailing slash.